src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -618,6 +618,119 @@ public virtual HttpNotFoundResult HttpNotFound()
         }
 
         /// <summary>
+        /// Creates an <see cref="BadRequestResult"/> that produces a Bad Request (400) response.
+        /// </summary>
+        /// <returns>The created <see cref="BadRequestResult"/> for the response.</returns>
+        [NonAction]
+        public virtual BadRequestResult HttpBadRequest()
+        {
+            return new BadRequestResult();
+        }
+
+        /// <summary>
+        /// Creates a <see cref="CreatedResult"/> object that produces a Created (201) response.
+        /// </summary>
+        /// <param name="uri">The URI at which the content has been created.</param>
+        /// <param name="value">The content value to negotiate and format in the entity body.</param>
+        /// <returns>The created <see cref="CreatedResult"/> for the response.</returns>
+        [NonAction]
+        public virtual CreatedResult Created([NotNull] string uri, object value)
+        {
+            return new CreatedResult(uri, value);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="CreatedResult"/> object that produces a Created (201) response.
+        /// </summary>
+        /// <param name="uri">The URI at which the content has been created.</param>
+        /// <param name="value">The content value to negotiate and format in the entity body.</param>
+        /// <returns>The created <see cref="CreatedResult"/> for the response.</returns>
+        [NonAction]
+        public virtual CreatedResult Created([NotNull] Uri uri, object value)
+        {
+            return new CreatedResult(uri.AbsoluteUri, value);
```

Comment:
What does WebAPI do if this is a relative URI? does it throw? I think we had special handling for this case.